tests/integration/modifiers/real-did-intersect-test.js

```diff
@@ -0,0 +1,48 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module(
+  'Integration | Modifier | did-intersect without mocks',
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    hooks.beforeEach(function () {
+      this.enterStub = sinon.stub();
+      this.exitStub = sinon.stub();
+      this.maxEnter = 1;
+      this.maxExit = 1;
+
+      this.observerManagerService = this.owner.lookup(
+        'service:ember-scroll-modifiers@observer-manager',
+      );
+    });
+
+    test('it removes elements from the registry on modifier cleanup', async function (assert) {
+      let removedElementsCount = 0;
+      let originalRemoveElement = this.observerManagerService.removeElement;
+      const observerManagerService = this.observerManagerService;
+
+      observerManagerService.removeElement = function removeElementExt(
+        ...args
+      ) {
+        removedElementsCount++;
+        originalRemoveElement.call(observerManagerService, ...args);
+      };
+      for (let index = 0; index < 50; index++) {
+        await render(
+          hbs`<div {{did-intersect onEnter=this.enterStub onExit=this.exitStub}}>
+          <span>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</span>
+        </div>`,
+        );
+      }
+
+      assert.strictEqual(
+        removedElementsCount,
+        49 * 2, // once for the element, once for a window object
```

Comment:
We should also assert against the underlying IntersectionObserverAdmin to ensure its registry gets cleaned up and prove that your implementation of `removeElement` works.

Reply:
Initially I wanted to inspect the count of elements, but the registry is a WeakMap which doesn't have a length/count method.

Reply:
I wonder if you could do something clever like grab a reference to one of the elements (which are the keys to the registry, I think?) and then assert that a `get()` call on the WeakMap returns undefined after it's been cleaned up? We wouldn't necessarily need to assert all of them, you could set up a separate test that just renders once and make sure that one gets cleaned up.

Reply:
I did think about that but then the issue is that you'd probably want to make sure that you no longer hold this reference yourself, so it'd just get funky :dancers:

Reply:
What I'm thinking is that you grab the reference only in the test. The sinon spy for `removeElement` should be able to remember what its call arguments were, which you could use to get access to the element, and then you can assert against the registry contents from there.